zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -521,23 +521,45 @@ def convert_slack_workspace_messages(slack_data_dir: str, users: List[ZerverFiel
 
     logging.info('######### IMPORTING MESSAGES STARTED #########\n')
 
-    message_json = {}
-    zerver_message, zerver_usermessage, attachment, uploads, \
-        reactions = channel_message_to_zerver_message(realm_id, users, added_users,
-                                                      added_recipient, all_messages,
-                                                      zerver_realmemoji,
-                                                      realm['zerver_subscription'],
-                                                      added_channels,
-                                                      domain_name)
-
-    message_json['zerver_message'] = zerver_message
-    message_json['zerver_usermessage'] = zerver_usermessage
-    message_filename = os.path.join(output_dir, "messages-000001.json")
-    logging.info("Writing Messages to %s\n" % (message_filename,))
-    write_data_to_file(os.path.join(message_filename), message_json)
+    total_reactions = []  # type: List[ZerverFieldsT]
+    total_attachments = []  # type: List[ZerverFieldsT]
+    total_uploads = []  # type: List[ZerverFieldsT]
+
+    message_id = usermessage_id = reaction_id = attachment_id = 0
+    id_list = [message_id, usermessage_id, reaction_id, attachment_id]
+
+    # The messages are stored in batches
+    low_index = 0
+    upper_index = low_index + chunk_size
+    dump_file_id = 1
+
+    while True:
+        message_json = {}
+        message_data = all_messages[low_index: upper_index]
+        if len(message_data) == 0:
+            break
+        zerver_message, zerver_usermessage, attachment, uploads, \
+            reactions, id_list = channel_message_to_zerver_message(
+                realm_id, users, added_users, added_recipient, message_data,
+                zerver_realmemoji, realm['zerver_subscription'], added_channels,
+                id_list, domain_name)
+
+        message_json['zerver_message'] = zerver_message
+        message_json['zerver_usermessage'] = zerver_usermessage
+        message_filename = os.path.join(output_dir, "messages-%06d.json" % (dump_file_id,))
+        logging.info("Writing Messages to %s\n" % (message_filename,))
+        write_data_to_file(os.path.join(message_filename), message_json)
+
+        low_index = upper_index
```

Comment:
I would always put these 'loop control' statements at the far end (or right at the top) to keep the looped code together between them.